air/src/air/context.rs

```diff
@@ -232,12 +241,16 @@ impl<B: StarkField> AirContext<B> {
 
     /// Returns the index of the auxiliary column which implements the Lagrange kernel, if any
     pub fn lagrange_kernel_aux_column_idx(&self) -> Option<usize> {
-        self.lagrange_kernel_aux_column_idx
+        if self.uses_logup_gkr() {
+            Some(self.trace_info().aux_segment_width() - 1)
+        } else {
+            None
+        }
     }
 
-    /// Returns true if the auxiliary trace segment contains a Lagrange kernel column
-    pub fn has_lagrange_kernel_aux_column(&self) -> bool {
-        self.lagrange_kernel_aux_column_idx().is_some()
+    /// Returns true if LogUp-GKR is enabled.
+    pub fn uses_logup_gkr(&self) -> bool {
+        self.logup_gkr
     }
```

Comment:
nit (feel free to ignore): maybe `logup_gkr_enabled()` would be a better name?

Reply:
I like this name better, updated!